### ISSUE TITLE
[Feature]: Detect duplicated images files during upload

### ISSUE THREAD

**Issue by febo** [metaplex-foundation/sugar#301]
### Feature

In some cases, some (several) assets might be associated with the same image file. Currently, this would lead to multiple upload (and charges) for the same file.

### Ideal solution or implementation

The `upload` command should maintain a list of assets already uploaded and before uploading a new one, check whether is has already been uploaded  or not. This can be done by comparing the hash of the image file, so even if image names are different – likely the case since they are named after the index of the asset – they won't be re-uploaded.

**Comment by Sonicfury** [comment]
Yes, especially that if using Shadow Drive, the file is accesible with its own name

**Comment by AlexITC** [comment]
Right now, there is an imaged required for each json file, for example, having these assets won't work:

- 0.json
- 0.jpg
- 1.json (linking 0.jpg)
